impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -178,7 +200,10 @@ static constexpr VkMemoryPropertyFlags ToVKMemoryPropertyFlags(
     case StorageMode::kDevicePrivate:
       return VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT;
     case StorageMode::kDeviceTransient:
-      return VK_MEMORY_PROPERTY_LAZILY_ALLOCATED_BIT;
+      if (supports_lazy_memory) {
```

Comment:
Looking at the memory types on vulkan.gpuinfo.org. We should be `&`ing these to be both device local and lazily allocated (if supported).